### PR TITLE
remove error handling from counter factual

### PR DESCRIPTION
**Pull request by xuke444**
Signed-off-by: Ke Xu <xuke@microsoft.com>

### CODE REVIEW

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 1]
```diff
@@ -185,48 +185,31 @@ def compute(self):
...
-                except Exception as e:
-                    cf_config.has_computation_failed = True
-                    cf_config.failure_reason = str(e)
-
-    def get(self, failed_to_compute=False):
-        """Return the computed counterfactual examples objects or failure reason.
-
-        :param failed_to_compute: Get the failure reasons if counterfactual
-                                  examples failed to compute.
-        :type failed_to_compute: bool
```
Comment (gaugup) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:212]: Why is the error handling removed? This handles the error scenario from dice-ml library and records the reason within the counterfactual manager.

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 2]
```diff
@@ -185,48 +185,31 @@ def compute(self):
...
-                        counterfactual_config.counterfactual_obj)
-            return counterfactual_obj_list
-        else:
-            failure_reason_list = []
-            for counterfactual_config in self._counterfactual_config_list:
-                if counterfactual_config.is_computed and \
-                        counterfactual_config.has_computation_failed:
-                    failure_reason_list.append(
-                        counterfactual_config.failure_reason)
-            return failure_reason_list
```
Comment (gaugup) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:229]: The get() method returns the counterfactual object or the reson why the computation failed. Why this removed?

**Review by gaugup — CHANGES_REQUESTED** [review]

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 1 above)
Comment (xuke444) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:212]: none of other manger is doing this.
This way is pretty confused to end user
We should line up with other manger throw error during compute.

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 2 above)
Comment (xuke444) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:229]: This is the issue all the error get hidden with this get, user will not notice when the counter factual fail

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:212]: Sorry, what is confusing to the other users? We can add a warning to signal to the user that the computation failed. 

If the other managers don't handle the error handling , they should address that issue.

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:229]: As I mentioned in the above comment, we could throw a warning that the computation failed as some reason. The error is not hidden. The error message is recorded in the class and user can access this error message. I am not able to understand in what capacity in the user not able to know that the counterfactual computation failed.

**Review by gaugup — CHANGES_REQUESTED** [review]
In my opinion, this PR removes the meaningful error handling during counterfactual examples and shouldn't me merged.

**Comment by gregorybchris** [comment]
I'm personally in favor of removing error handling like this. It's not very obvious to users how to retrieve error messages from failed analyses. In Python it is more common to raise an exception and let the caller handle the error than have a two step process of checking for a null result and then fetching the error for the failed computation.

**Comment by gaugup** [comment]
> I'm personally in favor of removing error handling like this. It's not very obvious to users how to retrieve error messages from failed analyses. In Python it is more common to raise an exception and let the caller handle the error than have a two step process of checking for a null result and then fetching the error for the failed computation.

Why wasn't this a concern in the PR where the error handling was added https://github.com/microsoft/responsible-ai-widgets/pull/460?

**Pull request closed without merging**